kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -354,17 +356,16 @@ private static Optional<Metadata> assignColumnIdAndPhysicalName(
       newSchema = rewriteFieldIdsForIceberg(newSchema, maxColumnId);
     }
 
+    // The maxColumnId in the metadata may be out-of-date either due to field-id assignment
+    // performed in this fx, or due to connector adding new fields
+    boolean shouldUpdateMaxId =
+        TableConfig.COLUMN_MAPPING_MAX_COLUMN_ID.fromMetadata(metadata) != maxColumnId.get();
```

Comment:
I think it would probably be better to do something like `<` here.  I don't know the defaults but we should make sure this is safe if the key is missing.

Reply:
Default is 0 I double checked; I think due to the way we initialize maxColId it is impossible for it to be less than `TableConfig.COLUMN_MAPPING_MAX_COLUMN_ID.fromMetadata(metadata)` but I can update this it doesn't really matter to me

i.e. we set maxColId using max fx that includes `TableConfig.COLUMN_MAPPING_MAX_COLUMN_ID.fromMetadata(metadata)` as an input

Reply:
Yeah, I agree it is probably impossible given the way the current code is constructed to have them be not equal.  This is more of a guard against accidental regressions in the code.